lib/utils/job-utils/racadm-parser.js

```diff
@@ -0,0 +1,229 @@
+            var row = [];
+            //Example of row
+            //row[0] - ['elementName', 'Intel(R) Ethernet 10G X520 LOM - 00:8C:FA:F3:78:30']
+            //row[1] - ['fQDD', 'NIC.Embedded.1-1-1']
+            //row[2] - ['installationDate', '2015-11-26T06:54:17Z']
+            //row[3] - ['currentVersion', '16.5.0']
+            _.forEach(data, function(line){
+                var splitedLine = line.split('='),
+                    name = splitedLine[0].trim().replace(' ', '');
+                var lowerFirstChar = name.substr(0,1).toLowerCase().concat(name.substr(1));
+                row.push([lowerFirstChar, splitedLine[1].trim()]);
+            });
+            var deviceName = row[1][1].split(':')[0];
+            var deviceInfo = {
+                elementName: '',
+                FQDD: '',
+                installationDate: '',
+                currentVersion: '',
+                rollbackVersion: '',
+                availableVersion: ''
+            };
+            if (softwareInventory.hasOwnProperty(deviceName)){
+                deviceInfo = softwareInventory[deviceName];
+                if (deviceInfo.installationDate === 'NA'){
+                    deviceInfo.installationDate = row[2][1];
+                }
+                if (!deviceInfo[row[3][0]]){
+                    deviceInfo[row[3][0]] = row[3][1];
+                }
+            } else {
+                deviceInfo.elementName = row[0][1];
+                deviceInfo.FQDD = row[1][1];
+                deviceInfo.installationDate = row[2][1];
+                deviceInfo[row[3][0]] = row[3][1];
+            }
+            /*
+             if (softwareInventory.hasOwnProperty(deviceName)){
+             deviceInfo = softwareInventory[deviceName];
+             deviceInfo[row[3][0]] = row[3][1];
+             if (deviceInfo[row[2][0]] === 'NA'){
+             deviceInfo[row[2][0]] = row[2][1];
+             }
+             } else{
+             for(var j = 0; j < data.length; j += 1){
+             deviceInfo[row[j][0]] = row[j][1];
+             }
+             }
+             */
+            softwareInventory[deviceName] = deviceInfo;
+        });
+        return this.simplifyKeyName(softwareInventory);
+    };
+
+    /**
+     * Create simply key for each software list
+     *
+     * @param {string} softwareList
+     */
+    RacadmCommandParser.prototype.simplifyKeyName = function(softwareList) {
+        var newKeyArray = [], oldKeyArray = [],
+            suffixArray = [], preffixArray = [], newSoftwareList = {};
+        //Original key                                      => New key
+        //Disk.Bay.0:Enclosure.Internal.0-0:RAID.Slot.1-1   => Disk0
+        //RAID.Slot.1-1                                     => RAID
+        _.forEach(softwareList, function(value, key){
+            preffixArray.push(key.toString().split('.')[0].trim());
+            suffixArray.push(key.toString().split(':')[0].split('.')[2]);
```

Comment:
If only pick the first or last splitted element, it better to use following code, this should get a little performance improvement:
```javascript
key.slice(0, key.indexOf('.'));   #get first splitted element
key.slice(key.lastIndexOf(':') + 1);  #get last splitted element
```